core/src/main/java/com/linecorp/armeria/server/cors/AbstractCorsPolicyBuilder.java

```diff
@@ -271,6 +277,37 @@ public AbstractCorsPolicyBuilder allowRequestMethods(Iterable<HttpMethod> method
         return this;
     }
 
+    /**
+     * Enables to allow all HTTP headers.
+     *
+     * <p>The server will set the CORS {@code "Access-Control-Allow-Headers"} to be as same as the CORS
+     * {@code "Access-Control-Request-Headers"} header in the request.
+     *
+     * @return {@code this} to support method chaining.
+     */
+    public AbstractCorsPolicyBuilder allowAllRequestHeaders() {
+        return allowAllRequestHeaders(false);
+    }
+
+    /**
+     * Enables to allow all HTTP headers.
+     *
+     * <p>If {@code useWildcard}, the server will set the CORS {@code "Access-Control-Allow-Headers"} to
+     * {@code "*"}. Please be careful since wildcard may have compatibility issue for some browsers.
+     *
+     * <p>If not {@code useWildcard}, the server will set the CORS {@code "Access-Control-Allow-Headers"} to be
+     * as same as the CORS {@code "Access-Control-Request-Headers"} header in the request.
+     *
+     * @return {@code this} to support method chaining.
+     */
+    public AbstractCorsPolicyBuilder allowAllRequestHeaders(boolean useWildcard) {
```

Comment:
After reading the [spec](developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers), wildcard does not actually mean accept all headers.
```
The value "*" only counts as a special wildcard value for requests without credentials (requests without HTTP cookies or HTTP authentication information).
In requests with credentials, it is treated as the literal header name "*" without special semantics.
Note that the Authorization header can't be wildcarded and always needs to be listed explicitly.
```
How about removing this option for less confusion. Users can easily use `*` expression using this builder that might not be inconvienent.

Reply:
Sure.

Reply:
I removed `useWildcard` option. Meanwhile, removed checking wildcard in `allowedRequestHeaders` to remain `CorsService` works as usual. So following two comments are outdated: https://github.com/line/armeria/pull/3843#discussion_r718334125 and https://github.com/line/armeria/pull/3843#discussion_r718279822.
If you think checking was better, let me recover it.